README.md
[![NPM version](https://badge.fury.io/js/cdk-efs-assets.svg)](https://badge.fury.io/js/cdk-efs-assets)
[![PyPI version](https://badge.fury.io/py/cdk-efs-assets.svg)](https://badge.fury.io/py/cdk-efs-assets)
![Release](https://github.com/pahud/cdk-efs-assets/workflows/Release/badge.svg)

# cdk-efs-assets

CDK construct library to populate Amazon EFS assets from Github or S3.

# `GithubSourceSync`

The `GithubSourceSync` deploys your Amazon EFS assets from specified Github repository. 

## Sample

```ts
import { GithubSourceSync } from 'cdk-efs-assets';

const app = new App();

const env = {
  region: process.env.CDK_DEFAULT_REGION ?? AWS_DEFAULT_REGION,
  account: process.env.CDK_DEFAULT_ACCOUNT,
};

const stack = new Stack(app, 'testing-stack', { env });

const vpc = ec2.Vpc.fromLookup(stack, 'Vpc', { isDefault: true })

const fs = new efs.FileSystem(stack, 'Filesystem', {
  vpc,
  removalPolicy: RemovalPolicy.DESTROY,
})

const efsAccessPoint = fs.addAccessPoint('EfsAccessPoint', {
  path: '/demo',
  createAcl: {
    ownerGid: '1001',
    ownerUid: '1001',
    permissions: '0755',
  },
  posixUser: {
    uid: '1001',
    gid: '1001',
  }
});

// create the one-time sync from Github repository to Amaozn EFS
new GithubSourceSync(stack, 'GithubSourceSync', {
  repository: 'https://github.com/pahud/cdk-efs-assets.git',
  efsAccessPoint,
  runsAfter: [fs.mountTargetsAvailable],
  vpc,
});

```

# `S3ArchiveSync`

The `S3ArchiveSync` deploys your Amazon EFS assets from a specified zip archive file stored in S3. The extracted contents will be placed into the root directory of the access point.

If the `syncOnUpdate` property is set to `true` (defaults to `true`), then the specified zip file path will be monitored, and if a new object is uploaded to the path, then it will resync the data to EFS. Note that to use this functionality, you must have a CloudTrail Trail in your account that captures the desired S3 write data event.

*WARNING*: The contents of the access point will be removed before extracting the zip file.

## Sample

```ts
import { S3ArchiveSync } from 'cdk-efs-assets';

const app = new App();

const env = {
  region: process.env.CDK_DEFAULT_REGION ?? AWS_DEFAULT_REGION,
  account: process.env.CDK_DEFAULT_ACCOUNT,
};

const stack = new Stack(app, 'testing-stack', { env });

const vpc = ec2.Vpc.fromLookup(stack, 'Vpc', { isDefault: true })

const fs = new efs.FileSystem(stack, 'Filesystem', {
  vpc,
  removalPolicy: RemovalPolicy.DESTROY,
})

const efsAccessPoint = fs.addAccessPoint('EfsAccessPoint', {
  path: '/demo',
  createAcl: {
    ownerGid: '1001',
    ownerUid: '1001',
    permissions: '0755',
  },
  posixUser: {
    uid: '1001',
    gid: '1001',
  }
});

const bucket = Bucket.fromBucketName(this, 'Bucket', 'demo-bucket');

// Will sync initial data from compressed S3 archive to EFS, and resync if the zip file in S3 changes
new S3ArchiveSync(this, 'S3ArchiveSync', {
  bucket,
  zipFilePath: 'folder/foo.zip',
  vpc,
  efsAccessPoint,
  runsAfter: [fs.mountTargetsAvailable],
});
```

# `S3SourceSync`

TBD




CHANGELOG.md
# Changelog

All notable changes to this project will be documented in this file. See [standard-version](https://github.com/conventional-changelog/standard-version) for commit guidelines.

### [0.1.75](https://github.com/pahud/cdk-efs-assets/compare/v0.1.74...v0.1.75) (2021-01-18)

### [0.1.74](https://github.com/pahud/cdk-efs-assets/compare/v0.1.73...v0.1.74) (2021-01-16)

### [0.1.73](https://github.com/pahud/cdk-efs-assets/compare/v0.1.72...v0.1.73) (2021-01-15)

### [0.1.72](https://github.com/pahud/cdk-efs-assets/compare/v0.1.71...v0.1.72) (2021-01-14)

### [0.1.71](https://github.com/pahud/cdk-efs-assets/compare/v0.1.70...v0.1.71) (2021-01-13)

### [0.1.70](https://github.com/pahud/cdk-efs-assets/compare/v0.1.69...v0.1.70) (2021-01-12)

### [0.1.69](https://github.com/pahud/cdk-efs-assets/compare/v0.1.68...v0.1.69) (2021-01-11)

### [0.1.68](https://github.com/pahud/cdk-efs-assets/compare/v0.1.67...v0.1.68) (2021-01-10)

### [0.1.67](https://github.com/pahud/cdk-efs-assets/compare/v0.1.66...v0.1.67) (2021-01-09)

### [0.1.66](https://github.com/pahud/cdk-efs-assets/compare/v0.1.65...v0.1.66) (2021-01-08)

### [0.1.65](https://github.com/pahud/cdk-efs-assets/compare/v0.1.64...v0.1.65) (2021-01-07)

### [0.1.64](https://github.com/pahud/cdk-efs-assets/compare/v0.1.63...v0.1.64) (2021-01-06)

### [0.1.63](https://github.com/pahud/cdk-efs-assets/compare/v0.1.62...v0.1.63) (2021-01-05)

### [0.1.62](https://github.com/pahud/cdk-efs-assets/compare/v0.1.61...v0.1.62) (2021-01-05)

### [0.1.61](https://github.com/pahud/cdk-efs-assets/compare/v0.1.60...v0.1.61) (2021-01-04)

### [0.1.60](https://github.com/pahud/cdk-efs-assets/compare/v0.1.59...v0.1.60) (2021-01-03)

### [0.1.59](https://github.com/pahud/cdk-efs-assets/compare/v0.1.58...v0.1.59) (2021-01-02)

### [0.1.58](https://github.com/pahud/cdk-efs-assets/compare/v0.1.57...v0.1.58) (2021-01-01)

### [0.1.57](https://github.com/pahud/cdk-efs-assets/compare/v0.1.56...v0.1.57) (2020-12-31)

### [0.1.56](https://github.com/pahud/cdk-efs-assets/compare/v0.1.55...v0.1.56) (2020-12-30)

### [0.1.55](https://github.com/pahud/cdk-efs-assets/compare/v0.1.54...v0.1.55) (2020-12-29)

### [0.1.54](https://github.com/pahud/cdk-efs-assets/compare/v0.1.53...v0.1.54) (2020-12-29)

### [0.1.53](https://github.com/pahud/cdk-efs-assets/compare/v0.1.52...v0.1.53) (2020-12-29)

### [0.1.52](https://github.com/pahud/cdk-efs-assets/compare/v0.1.51...v0.1.52) (2020-12-28)

### [0.1.51](https://github.com/pahud/cdk-efs-assets/compare/v0.1.50...v0.1.51) (2020-12-27)

### 0.1.50 (2020-12-21)

### 0.1.49 (2020-12-20)

### 0.1.48 (2020-12-19)

### 0.1.47 (2020-11-24)

### 0.1.46 (2020-11-23)

### 0.1.45 (2020-11-22)

### 0.1.44 (2020-11-18)

### 0.1.43 (2020-11-17)

### 0.1.42 (2020-11-16)

### 0.1.41 (2020-11-15)

### 0.1.40 (2020-11-14)

### 0.1.39 (2020-11-13)

### 0.1.38 (2020-11-12)

### 0.1.37 (2020-11-11)

### 0.1.36 (2020-11-10)

### 0.1.35 (2020-11-09)

### 0.1.34 (2020-11-08)

### 0.1.33 (2020-11-07)

### 0.1.32 (2020-11-06)

### 0.1.31 (2020-11-05)

### 0.1.30 (2020-11-04)

### 0.1.29 (2020-11-03)

### 0.1.28 (2020-11-02)

### 0.1.27 (2020-11-01)

### 0.1.26 (2020-11-01)

### 0.1.25 (2020-10-30)

### 0.1.24 (2020-10-29)

### 0.1.23 (2020-10-28)

### 0.1.22 (2020-10-27)

### 0.1.21 (2020-10-26)

### 0.1.20 (2020-10-25)

### 0.1.19 (2020-10-24)

### 0.1.18 (2020-10-23)

### 0.1.17 (2020-10-22)

### 0.1.16 (2020-10-21)

### 0.1.15 (2020-10-20)

### 0.1.14 (2020-10-19)

### 0.1.13 (2020-10-18)

### 0.1.12 (2020-10-17)

### 0.1.11 (2020-10-16)

### 0.1.10 (2020-10-15)

### 0.1.9 (2020-10-14)

### 0.1.8 (2020-10-13)

### [0.1.7](https://github.com/pahud/cdk-efs-assets/compare/v0.1.6...v0.1.7) (2020-10-12)

### 0.1.6 (2020-10-12)

### [0.1.5](https://github.com/pahud/cdk-efs-assets/compare/v0.1.4...v0.1.5) (2020-10-12)

### [0.1.4](https://github.com/pahudnet/cdk-efs-assets/compare/v0.1.2...v0.1.4) (2020-10-12)

### [0.1.2](https://github.com/pahudnet/cdk-efs-assets/compare/v0.1.0...v0.1.2) (2020-10-12)

### [0.0.2](https://github.com/pahudnet/cdk-efs-assets/compare/v0.0.1...v0.0.2) (2020-10-12)

### 0.0.2 (2020-10-12)

### 0.0.1 (2020-10-12)


.projenrc.js
const {
  AwsCdkConstructLibrary,
} = require('projen');

const AUTOMATION_TOKEN = 'AUTOMATION_GITHUB_TOKEN';

const project = new AwsCdkConstructLibrary({
  authorAddress: 'pahudnet@gmail.com',
  authorName: 'Pahud Hsieh',
  cdkVersion: '1.77.0',
  name: 'cdk-efs-assets',
  repository: 'https://github.com/pahud/cdk-efs-assets.git',
  description: 'Amazon EFS assets from Github repositories or S3 buckets',
  cdkDependencies: [
    '@aws-cdk/core',
    '@aws-cdk/aws-ec2',
    '@aws-cdk/aws-efs',
    '@aws-cdk/aws-iam',
    '@aws-cdk/aws-lambda',
    '@aws-cdk/aws-secretsmanager',
    '@aws-cdk/custom-resources',
  ],
  dependabot: false,

  keywords: [
    'aws',
    'cdk',
    'efs',
    'github',
  ],

  catalog: {
    twitter: 'pahudnet',
    announce: false,
  },

  python: {
    distName: 'cdk-efs-assets',
    module: 'cdk_efs_assets',
  },
});

// create a custom projen and yarn upgrade workflow
const workflow = project.github.addWorkflow('ProjenYarnUpgrade');

workflow.on({
  schedule: [{
    cron: '11 0 * * *',
  }], // 0:11am every day
  workflow_dispatch: {}, // allow manual triggering
});

workflow.addJobs({
  upgrade: {
    'runs-on': 'ubuntu-latest',
    'steps': [
      { uses: 'actions/checkout@v2' },
      {
        uses: 'actions/setup-node@v1',
        with: {
          'node-version': '10.17.0',
        },
      },
      { run: 'yarn upgrade' },
      { run: 'yarn projen:upgrade' },
      // submit a PR
      {
        name: 'Create Pull Request',
        uses: 'peter-evans/create-pull-request@v3',
        with: {
          'token': '${{ secrets.' + AUTOMATION_TOKEN + ' }}',
          'commit-message': 'chore: upgrade projen',
          'branch': 'auto/projen-upgrade',
          'title': 'chore: upgrade projen and yarn',
          'body': 'This PR upgrades projen and yarn upgrade to the latest version',
          'labels': 'auto-merge',
        },
      },
    ],
  },
});


const common_exclude = ['cdk.out', 'cdk.context.json', 'images', 'yarn-error.log'];
project.npmignore.exclude(...common_exclude);
project.gitignore.exclude(...common_exclude);

project.synth();


src/index.ts
import * as path from 'path';
import * as ec2 from '@aws-cdk/aws-ec2';
import * as efs from '@aws-cdk/aws-efs';
import * as lambda from '@aws-cdk/aws-lambda';
import * as cdk from '@aws-cdk/core';
import * as cr from '@aws-cdk/custom-resources';

export interface GithubSourceFeederProps {
  /**
   * The target Amazon EFS filesystem to clone the github repository to.
   */
  readonly efsAccessPoint: efs.AccessPoint;
  /**
   * The VPC of the Amazon EFS Filesystem.
   */
  readonly vpc: ec2.IVpc;
  /**
   * The github repository HTTP URI.
   */
  readonly repository: string;
  /**
   * The dependent resources before triggering the sync.
   */
  readonly runsAfter?: cdk.IDependable[];
}

export class GithubSourceSync extends cdk.Construct {
  constructor(scope: cdk.Construct, id: string, props: GithubSourceFeederProps ) {
    super(scope, id);

    const stack = cdk.Stack.of(this);
    const region = stack.region;

    const handler = new lambda.Function(this, 'Handler', {
      runtime: lambda.Runtime.PYTHON_3_8,
      code: lambda.Code.fromAsset(path.join(__dirname, '../lambda-handler')),
      handler: 'index.on_event',
      layers: [
        lambda.LayerVersion.fromLayerVersionArn(this, 'GitLayer', `arn:aws:lambda:${region}:553035198032:layer:git-lambda2:7`),
      ],
      filesystem: lambda.FileSystem.fromEfsAccessPoint(props.efsAccessPoint, '/mnt/efsmount'),
      vpcSubnets: {
        subnetType: ec2.SubnetType.PRIVATE,
      },
      vpc: props.vpc,
      memorySize: 512,
      timeout: cdk.Duration.minutes(1),
      environment: {
        REPOSITORY_URI: props.repository,
        MOUNT_TARGET: '/mnt/efsmount',
      },
      currentVersionOptions: {
        provisionedConcurrentExecutions: 1,
      },
    });

    // create a custom resource to trigger the sync
    const myProvider = new cr.Provider(this, 'MyProvider', {
      onEventHandler: handler,
    });

    new cdk.CustomResource(this, 'SyncTrigger', { serviceToken: myProvider.serviceToken });

    // ensure the dependency
    if (props.runsAfter) {
      handler.node.addDependency(...props.runsAfter);
      // triggerResource.node.addDependency(...props.runsAfter);
    };

    // handler.node.addDependency(props.efsAccessPoint);
    // triggerResource.node.addDependency(props.efsAccessPoint);
  }
}


lambda-handler/index.py
import json
import logging
import os
import subprocess

logger = logging.getLogger()
logger.setLevel(logging.INFO)

repo = os.environ.get('REPOSITORY_URI')
mount_target = os.environ.get('MOUNT_TARGET', '/mnt/efsmount')

def on_event(event, context):
  print(event)
  request_type = event['RequestType']
  if request_type == 'Create': return on_create(event)
  if request_type == 'Update': return on_update(event)
  if request_type == 'Delete': return on_delete(event)
  raise Exception("Invalid request type: %s" % request_type)

def on_create(event):
  props = event["ResourceProperties"]
  print("create new resource with props %s" % props)

  # add your create code here...
  # physical_id = ...

  sync(repo, mount_target)

  ok_result = {'status': 'ok'}

  return { 'Data': ok_result }

  # return { 'PhysicalResourceId': physical_id }

def on_update(event):
  physical_id = event["PhysicalResourceId"]
  props = event["ResourceProperties"]
  print("update resource %s with props %s" % (physical_id, props))
  # ...

def on_delete(event):
  physical_id = event["PhysicalResourceId"]
  print("delete resource %s" % physical_id)
  # ...


def sync(repo, target_path):
    subprocess.check_call([ 'git', 'clone', repo, mount_target ])


src/synced-access-point.ts
import * as path from 'path';
import * as ec2 from '@aws-cdk/aws-ec2';
import * as efs from '@aws-cdk/aws-efs';
import { LambdaFunction } from '@aws-cdk/aws-events-targets';
import { PolicyStatement } from '@aws-cdk/aws-iam';
import * as lambda from '@aws-cdk/aws-lambda';
import * as s3 from '@aws-cdk/aws-s3';
import * as cdk from '@aws-cdk/core';
import * as cr from '@aws-cdk/custom-resources';

export interface SyncSourceProps {
  /**
   * The VPC of the Amazon EFS Filesystem.
   */
  readonly vpc: ec2.IVpc;
  /**
   * Where to place the network interfaces within the VPC.
   */
  readonly vpcSubnets?: ec2.SubnetSelection;
  /**
   * Timeout duration for sync Lambda function. (optional, default: Duration.minutes(3))
   */
  readonly timeout?: cdk.Duration;
  /**
   * The (absolute) directory path inside the EFS AccessPoint to sync files to. Specify '/' to restore synced files to the root
   * directory. (optional, default: a source-specific directory path. For example, for the GitHub source, the default
   * behavior is to restore to a directory matching the name of the repository)
   */
  readonly syncDirectoryPath?: string;
}

export interface GithubSourceProps extends SyncSourceProps {
  /**
   * The github repository HTTP URI.
   */
  readonly repository: string;
}

export interface S3ArchiveSourceProps extends SyncSourceProps {
  /**
   * The S3 bucket containing the archive file.
   */
  readonly bucket: s3.IBucket;

  /**
   * The path of the zip file to extract in the S3 bucket.
   */
  readonly zipFilePath: string;

  /**
   * If this is set to true, then whenever a new object is uploaded to the specified path, an EFS sync will be triggered.
   * Currently, this functionality depends on at least one CloudTrail Trail existing in your account that captures the S3
   * event.
   *
   * (optional, default: true)
   */
  readonly syncOnUpdate?: boolean;
}

export abstract class SyncSource {
  public static github(props: GithubSourceProps): SyncSource {
    return new GithubSyncSource(props);
  }

  public static s3Archive(props: S3ArchiveSourceProps): SyncSource {
    return new S3ArchiveSyncSource(props);
  }

  /** @internal */
  abstract _createHandler(accessPoint: efs.AccessPoint): lambda.Function;
}

class GithubSyncSource extends SyncSource {
  private readonly props: GithubSourceProps;

  constructor(props: GithubSourceProps) {
    super();
    this.props = props;
  }

  _createHandler(accessPoint: efs.AccessPoint): lambda.Function {
    const stack = cdk.Stack.of(accessPoint);
    const region = stack.region;

    const vpcSubnets = this.props.vpcSubnets ?? { subnetType: ec2.SubnetType.PRIVATE };
    const timeout = this.props.timeout ?? cdk.Duration.minutes(3);

    let syncDirectoryPath;
    if (this.props.syncDirectoryPath === undefined) {
      // if property is unspecified, use repository name as output directory

      const splitDirs = this.props.repository.split('/');
      syncDirectoryPath = '/' + splitDirs[splitDirs.length - 1].split('.')[0];
    } else {
      syncDirectoryPath = this.props.syncDirectoryPath;
    }

    const handler = new lambda.Function(accessPoint, 'GithubHandler', {
      runtime: lambda.Runtime.PYTHON_3_8,
      code: lambda.Code.fromAsset(path.join(__dirname, '../lambda-handler', 'github-sync')),
      handler: 'index.on_event',
      layers: [
        lambda.LayerVersion.fromLayerVersionArn(
          accessPoint,
          'GitLayer',
          `arn:aws:lambda:${region}:553035198032:layer:git-lambda2:7`,
        ),
      ],
      filesystem: lambda.FileSystem.fromEfsAccessPoint(accessPoint, '/mnt/efsmount'),
      vpcSubnets: vpcSubnets,
      vpc: this.props.vpc,
      memorySize: 512,
      timeout: timeout,
      environment: {
        REPOSITORY_URI: this.props.repository,
        MOUNT_TARGET: '/mnt/efsmount',
        SYNC_PATH: syncDirectoryPath,
      },
      currentVersionOptions: {
        provisionedConcurrentExecutions: 1,
      },
    });

    return handler;
  }
}

class S3ArchiveSyncSource extends SyncSource {
  private readonly props: S3ArchiveSourceProps;

  constructor(props: S3ArchiveSourceProps) {
    super();
    this.props = props;
  }

  _createHandler(accessPoint: efs.AccessPoint): lambda.Function {
    const vpcSubnets = this.props.vpcSubnets ?? { subnetType: ec2.SubnetType.PRIVATE };
    const syncOnUpdate = this.props.syncOnUpdate ?? true;
    const timeout = this.props.timeout ?? cdk.Duration.minutes(3);

    const splitDirs = this.props.zipFilePath.split('/');
    const filename = splitDirs[splitDirs.length - 1].split('.')[0];

    let syncDirectoryPath;
    if (this.props.syncDirectoryPath === undefined) {
      // if property is unspecified, use zip file name as output directory

      syncDirectoryPath = '/' + filename;
    } else {
      syncDirectoryPath = this.props.syncDirectoryPath;
    }

    const handler = new lambda.Function(accessPoint, 'SyncHandler', {
      runtime: lambda.Runtime.PYTHON_3_8,
      code: lambda.Code.fromAsset(path.join(__dirname, '../lambda-handler', 's3-archive-sync')),
      handler: 'index.on_event',
      filesystem: lambda.FileSystem.fromEfsAccessPoint(accessPoint, '/mnt/efsmount'),
      vpcSubnets: vpcSubnets,
      vpc: this.props.vpc,
      memorySize: 512,
      timeout: timeout,
      environment: {
        MOUNT_TARGET: '/mnt/efsmount',
        BUCKET_NAME: this.props.bucket.bucketName,
        ZIPPED_KEY: this.props.zipFilePath,
        SYNC_PATH: syncDirectoryPath,
      },
      currentVersionOptions: {
        provisionedConcurrentExecutions: 1,
      },
      initialPolicy: [
        new PolicyStatement({
          actions: ['s3:GetObject*'],
          resources: ['arn:aws:s3:::' + this.props.bucket.bucketName + '/' + this.props.zipFilePath],
        }),
      ],
    });

    if (syncOnUpdate) {
      // In order to support bucket notifications for imported IBucket objects, onCloudTrailWriteObject is used.
      // TODO: When https://github.com/aws/aws-cdk/issues/2004 is closed, can use handler.addEventSource instead.
      /*
      handler.addEventSource(
        new S3EventSource(props.bucket, {
          events: [s3.EventType.OBJECT_CREATED],
          filters: [{ prefix: props.zipFilePath }]
        })
      );
       */

      this.props.bucket.onCloudTrailWriteObject('S3FileListener-' + filename, {
        paths: [this.props.zipFilePath],
        target: new LambdaFunction(handler),
      });
    }

    return handler;
  }
}

export interface SyncedAccessPointProps extends efs.AccessPointProps {
  readonly syncSource: SyncSource;
}

export class SyncedAccessPoint extends efs.AccessPoint implements efs.IAccessPoint {
  constructor(scope: cdk.Construct, id: string, props: SyncedAccessPointProps) {
    super(scope, id, props);

    const handler = props.syncSource._createHandler(this);

    // create a custom resource to trigger the sync
    const myProvider = new cr.Provider(this, 'Provider', {
      onEventHandler: handler,
    });

    new cdk.CustomResource(this, 'SyncTrigger', { serviceToken: myProvider.serviceToken });

    // ensure the mount targets are available as dependency for the sync function
    handler.node.addDependency(props.fileSystem.mountTargetsAvailable);
  }
}


src/integ.default.ts
import * as ec2 from '@aws-cdk/aws-ec2';
import * as efs from '@aws-cdk/aws-efs';
import { App, Stack, RemovalPolicy } from '@aws-cdk/core';
import { GithubSourceSync } from './';

const AWS_DEFAULT_REGION = 'us-east-1';

export class IntegTesting {
  readonly stack: Stack[];

  constructor() {
    const app = new App();

    const env = {
      region: process.env.CDK_DEFAULT_REGION ?? AWS_DEFAULT_REGION,
      account: process.env.CDK_DEFAULT_ACCOUNT ?? '11111111111',
    };

    const stack = new Stack(app, 'testing-stack9', { env });

    const vpc = ec2.Vpc.fromLookup(stack, 'Vpc', { isDefault: true });

    const fs = new efs.FileSystem(stack, 'Filesystem', {
      vpc,
      removalPolicy: RemovalPolicy.DESTROY,
    });

    const efsAccessPoint = fs.addAccessPoint('EfsAccessPoint', {
      path: '/demo',
      createAcl: {
        ownerGid: '1001',
        ownerUid: '1001',
        permissions: '0755',
      },
      posixUser: {
        uid: '1001',
        gid: '1001',
      },
    });

    // create the one-time sync from Github repository to Amaozn EFS
    new GithubSourceSync(stack, 'GithubSourceSync', {
      repository: 'https://github.com/pahud/cdk-efs-assets.git',
      efsAccessPoint,
      runsAfter: [fs.mountTargetsAvailable],
      vpc,
    });

    this.stack = [stack];
  };
}

// run the integ testing
new IntegTesting();
